xrpl/models/transactions/oracle_set.py

```diff
@@ -0,0 +1,48 @@
+"""Model for OracleSet transaction type."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+from xrpl.models.nested_model import NestedModel
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class OracleSet(Transaction):
+    """Represents a OracleSet transaction."""
+
+    account: str = REQUIRED  # type: ignore
+    oracle_document_id: int = REQUIRED  # type: ignore
+
+    """
+    The below three fields must be hex-encoded. You can
+    use `xrpl.utils.str_to_hex` to convert a UTF-8 string to hex.
+    """
+    provider: Optional[str] = None
+    uri: Optional[str] = None
+    asset_class: Optional[str] = None
+
+    last_update_time: int = REQUIRED  # type: ignore
+    price_data_series: List[PriceData] = REQUIRED  # type: ignore
+
+    transaction_type: TransactionType = field(
+        default=TransactionType.ORACLE_SET,
+        init=False,
+    )
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class PriceData(NestedModel):
+    """Represents one PriceData element. It is used in OracleSet transaction"""
+
+    base_asset: str = REQUIRED  # type: ignore
+    quote_asset: str = REQUIRED  # type: ignore
+    asset_price: Optional[int] = None
+    scale: Optional[int] = None
```

Comment:
These will need docstrings too or put a TODO to add docs later.

Reply:
fixed in [9c7ce5e](https://github.com/XRPLF/xrpl-py/pull/701/commits/9c7ce5e9e7cabdb5105c193447a2deb27bc6f59e)